### DISCUSSION TITLE
Clarification on `generate_core_metadata'

### DISCUSSION THREAD

**Discussion by tannhorn** [idaholab/moose#24558] [Q&A Meshing] [answered]
When using the reactor module, does it make sense to set ```generate_core_metadata = true``` if the core has only one level of patterning? The code is set to disallow ```generate_core_metadata = true``` with ```PatternedHexMeshGenerator``` object using ```PolygonConcentricCircleMeshGenerator``` objects as pattern inputs.

However, note that if using ```HexagonConcentricCircleAdaptiveBoundaryMeshGenerator``` instead of  ```PolygonConcentricCircleMeshGenerator``` (basically just replacing the type), core metadata can now be generated. Not sure if this working as intended, either.

**Comment by GiudGiud** [comment]
@miaoyinb

**Reply by miaoyinb** [reply, marked as answer]
`HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` is used to generate a mesh that contains assembly-style metadata, which can be used by `PatternedHexMeshGenerator` as input unit assembly mesh for a core mesh (`generate_core_metadata = true`). `HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` also provides the "adaptive boundary" feature, but you do not have to use it.

Mesh generated by `PolygonConcentricCircleMeshGenerator` only contains pin-style metadata, so it cannot be used as an input unit assembly mesh.

**Reply by tannhorn** [reply]
Thank you for the answer. However, I am still not sure if I need to generate core_metadata when my core is only singly-heterogeneous (i.e. only a lattice of pins, rather than a lattice of assemblies which themselves comprise of lattices of pins).
